t/re-split.t

```diff
@@ -0,0 +1,707 @@
+                return
+            end
+
+            for i = 1, #my_table do
+                ngx.say(res[i])
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+a
+,
+b
+,
+c,d,e
+--- error_log eval
+qr/\[TRACE   \d+/
+--- no_error_log
+[error]
+
+
+
+=== TEST 16: split matches, ctx arg
+--- http_config eval: $::HttpConfig
+--- config
+    location = /re {
+        content_by_lua_block {
+            local ngx_re = require "ngx.re"
+
+            local res, err = ngx_re.split("a,b,c,d,e", ",", nil, { pos = 5 })
+            if err then
+                ngx.log(ngx.ERR, "failed: ", err)
+                return
+            end
+
+            for i = 1, #res do
+                ngx.say(res[i])
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+c
+d
+e
+--- error_log eval
+qr/\[TRACE   \d+/
+--- no_error_log
+[error]
+
+
+
+=== TEST 17: split matches, trailing subjects
+--- http_config eval: $::HttpConfig
+--- config
+    location /re {
+        content_by_lua_block {
+            local ngx_re = require "ngx.re"
+
+            local res, err = ngx_re.split(",a,b,c,d,", ",")
+            if err then
+                ngx.log(ngx.ERR, "failed: ", err)
+                return
+            end
+
+            for i = 1, #res do
+                if res[i] == "" then
+                    ngx.say("_blank_")
+                else
+                    ngx.say(res[i])
+                end
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+_blank_
+a
+b
+c
+d
+_blank_
+--- error_log eval
+qr/\[TRACE   \d+/
+--- no_error_log
+[error]
+attempt to get length of local 'regex' (a number value)
+
+
+
+=== TEST 18: split matches, real use-case
+--- http_config eval: $::HttpConfig
+--- config
+    location /re {
+        content_by_lua_block {
+            local ngx_re = require "ngx.re"
+
+            local res, err = ngx_re.split("abcd,erfg,ghij;hello world;aaa", ",|;")
+            if err then
+                ngx.log(ngx.ERR, "failed: ", err)
+                return
+            end
+
+            for i = 1, #res do
+                ngx.say(res[i])
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+abcd
+erfg
+ghij
+hello world
+aaa
+--- error_log eval
+qr/\[TRACE   \d+/
+--- no_error_log
+[error]
+
+
+
+=== TEST 19: split no matches
+--- http_config eval: $::HttpConfig
+--- config
+    location /re {
+        content_by_lua_block {
+            local ngx_re = require "ngx.re"
+
+            local res, err = ngx_re.split("abcd", ",")
+            if err then
+                ngx.log(ngx.ERR, "failed: ", err)
+                return
+            end
+
+            for i = 1, #res do
+                ngx.say(res[i])
+            end
+        }
+    }
+--- request
+GET /re
+--- response_body
+abcd
+--- error_log eval
+qr/\[TRACE   \d+/
+--- no_error_log
+[error]
+
+
+
+=== TEST 20: subject is not a string type
+--- http_config eval: $::HttpConfig
```

Comment:
@thibaultcha Because you need to disable the nginx memory pool *and* the built-in allocator of LuaJIT to maximize the strength of valgrind. Please read the book section I referenced earlier in this PR for more details on this: https://openresty.gitbooks.io/programming-openresty/content/testing/test-modes.html#_valgrind_mode